CasadiFunctions/createCasadi_Ligaments.m

```diff
@@ -0,0 +1,178 @@
+    
+    end
+    
+    
+    %% Ligaments that cross multiple coordinates
+    idx_multi_coord = find(sum(model_info.ligament_info.ligament_spanning_multi_coord,2));
+
+    % Assemble polynomial approximation from coefficients
+    qin_SX = SX.sym('qin',model_info.ExtFunIO.jointi.nq.all,1);
+    l_lig_p_SX = SX(model_info.ligament_info.NLigament,1);
+
+    if sum(model_info.ligament_info.ligament_spanning_multi_coord,'all') > 0
+        ligament_spanning_info_m = model_info.ligament_info.ligament_spanning_multi_coord(:,:);
+        LigamentInfo_m.muscle = model_info.ligament_info.polyFit.LigamentInfo.muscle(:);
+        load nCoeffMat
+        load expoVal_all
+        for i=1:model_info.ligament_info.NLigament      
+            index_dof_crossing = find(ligament_spanning_info_m(i,:)==1);
+            nr_dof_crossing = length(index_dof_crossing);
+            if nr_dof_crossing == 0
+                continue
+            end
+            order = LigamentInfo_m.muscle(i).order;
+            [mat,~] = n_art_mat_3_cas_SX_7(qin_SX(index_dof_crossing,1)',...
+                nCoeffMat(order,nr_dof_crossing), expoVal_all{order,nr_dof_crossing});
+            l_lig_p_SX(i,1) = mat'*LigamentInfo_m.muscle(i).coeff; 
+        end 
+    end
+    f_l_lig = Function('f_l_lig',{qin_SX},{l_lig_p_SX});
+    
+    % Create casadi function force-length
+    l_lig_SX = SX.sym('l_lig',model_info.ligament_info.NLigament,1);
+    F_lig_SX = SX(model_info.ligament_info.NLigament,1);
+    for j=idx_multi_coord'
+        % test for valid input force-length
+        if ~exist(model_info.ligament_info.parameters(j).stiffness,'file')
+            error(['Unable to find ' model_info.ligament_info.parameters(j).stiffness,...
+                '. Add the function to ./ModelComponents, or select an existing function for ',...
+                'the force-length of ligament ' model_info.ligament_info.parameters(j).ligament_name '.'])
+        end
+        % evaluate force-length
+        f_stiffness = str2func(model_info.ligament_info.parameters(j).stiffness);
+        CSA = model_info.ligament_info.parameters(j).cross_section_area;
+        ls = model_info.ligament_info.parameters(j).slack_length;
+        F_lig_SX(j) = f_stiffness(CSA,ls,l_lig_SX(j));
+    end
+    
+    f_F_lig = Function('f_F_lig',{l_lig_SX},{F_lig_SX});
+    
+    % Calculate moments
+    l_lig_MX = f_l_lig(Qs_MX);
+    F_lig_MX = f_F_lig(l_lig_MX);
+    % transposed Jacobian of lengths to angles (i.e. moment arms), multiplied by forces
+    M_lig_multi_MX = -jtimes(l_lig_MX,Qs_MX,F_lig_MX,true); 
+
+    % Lengths and forces for post-processing
+    L_lig_pp_MX = L_lig_pp_MX + l_lig_MX; % both vectors cannot have a non-zero on the same index.
+    F_lig_pp_MX = F_lig_pp_MX + F_lig_MX; 
+    
+    v_lig_pp_MX = jtimes(L_lig_pp_MX,Qs_MX,Qdots_MX);
+
+else
+    % if there are no ligaments, moments are zero (sparse)
+    M_lig_multi_MX = MX(model_info.ExtFunIO.jointi.nq.all,1);
```

Comment:
I think that you have to do something similar for M_lig_single_MX

Reply:
M_lig_single_MX is initialised with zeros at the beginning. I added initialisation for M_lig_multi_MX as well, instead of the else-statement. This is funtionally the same, but should be more clear.